src/shared/components/exonsCharts/ExonsChartStore.ts

```diff
@@ -0,0 +1,242 @@
+
+    getExonsBySite(siteId: number, transcriptId: string, breakpoint: number): Array<ExonRangeExt> {
+        let exons: Array<ExonRangeExt> = [];
+        if (siteId === 1 || siteId === 2) {
+            let _t = this.getTranscriptsById(transcriptId)[0];
+            let _sortedExons = <ExonRangeExt[]> _t.exons.sort(
+                // sort exons by rank
+                (exon1: ExonRangeExt, exon2: ExonRangeExt) => exon1.rank - exon2.rank
+            );
+            if (siteId === 1) {
+                exons = _sortedExons.filter(exon => exon.rank <= breakpoint);
+            } else if (siteId === 2) {
+                exons = _sortedExons.filter(exon => exon.rank >= breakpoint);
+            }
+        }
+        return exons;
+    }
+
+    getTranscriptsById(transcriptId: string): Array<EnsemblTranscriptExt> {
+        return (this.transcripts.result || []).filter(d => {
+            return d.transcriptId === transcriptId;
+        });
+    }
+
+    @computed
+    get computedFusions(): StructuralVariantExt[] {
+        return this.fusionDataStore.tableData.map((f: StructuralVariantExt[]) => {
+            let site1Exons = this.getExonsBySite(
+                1,
+                f[0].site1EnsemblTranscriptId,
+                f[0].site1Exon
+            );
+            let site2Exons = this.getExonsBySite(
+                2,
+                f[0].site2EnsemblTranscriptId,
+                f[0].site2Exon
+            );
+            f[0].isLeftAligned = f[0].site1HugoSymbol === this.gene.hugoGeneSymbol;
+            f[0].exons = site1Exons.concat(site2Exons);
+            f[0].totalWidth = ExonsChartStore.getTotalWidth(f[0].exons || []);
+            return f[0];
+        });
+    }
+
+    /**
+     * Get total width of exons.
+     * @param {ExonRangeExt[]} exons
+     * @returns {number} return 0 when no width defined.
+     */
+    static getTotalWidth(exons: ExonRangeExt[]): number {
+        return exons.reduce((accumulator, e) => {
+            e.width = e.width ? e.width : 0;
+            return accumulator + e.width;
+        }, 0);
+    }
+
+    @computed
+    get computedTranscripts(): EnsemblTranscriptExt[] {
+        return (this.transcripts.result || [])
+            .map(t => {
+                t.exons = <ExonRangeExt[]> t.exons
+                    .sort(
+                        // sort exons by rank
+                        (exon1: ExonRangeExt, exon2: ExonRangeExt) => exon1.rank - exon2.rank
+                    )
+                    .map((e: ExonRangeExt) => {
+                        // add color
+                        e.fillColor = e.fillColor ? e.fillColor : t.fillColor;
+                        // calculate width of exon
+                        e.width = e.exonEnd - e.exonStart;
+                        return e;
+                    });
+                t.totalWidth = ExonsChartStore.getTotalWidth(<ExonRangeExt[]> t.exons);
+                t.deltaX = 0; // initialise delta for the x axis
```

Comment:
you're modifying the actual object transcripts.result

what you're probably trying to do is:

`return Object.assign({}, t, { totalWidth: ..., deltaX: ... });`